ui/app/utils/tensorzero/tensorzero.ts

```diff
@@ -594,15 +557,32 @@ export class TensorZeroClient {
     return await fetch(url, { method, headers, body });
   }
 
+  private async getErrorText(response: Response): Promise<string> {
+    if (response.bodyUsed) {
+      response = response.clone();
+    }
+    const responseText = await response.text();
+    try {
+      const parsed = JSON.parse(responseText);
+      return typeof parsed?.error === "string" ? parsed.error : responseText;
+    } catch {
+      // Invalid JSON; return plain text from response
+      return responseText;
+    }
+  }
+
   private handleHttpError({
     message,
     response,
   }: {
     message: string;
     response: Response;
   }): never {
-    // TODO: Switch on HTTP status code to handle known errors with more
-    // specificity
-    throw new HttpError({ message, response });
+    throw new TensorZeroServerError(message, {
+      // TODO: Ensure that server errors do not leak sensitive information to
```

Comment:
Afaik we make sure not to leak sensitive info in errors already as long as we're not in debug mode. I think this is OK to do.